packages/create-apollo-client/src/createApolloLinks.js

```diff
@@ -0,0 +1,207 @@
+// @flow
+
+import * as R from 'ramda';
+import { ApolloLink } from 'apollo-link';
+import { BatchHttpLink } from 'apollo-link-batch-http';
+import { onError } from 'apollo-link-error';
+import { FileUploadLink, TokenRefreshLink } from '@8base/sdk';
+
+
+/**
+ * Interface of the user auth info. Needs for the auth links correctly working.
+ */
+interface AuthState {
+  email: string,
+  organizationId: string,
+  accountId: string,
+  idToken: string,
+  refreshToken: string,
+}
+
+/**
+ * Interface of the Graphql error which arise on the server side.
+ */
+interface GraphQLError {
+  code: string,
+  details: Object,
+  message: string,
+}
+
+type GetErrorLinkParams = {
+  onGraphQLErrors?: (GraphQLError[]) => void,
+  onNetworkError?: (error: Object) => void,
+}
+
+type GetAuthStateParams = {
+  getAuthState?: () => AuthState,
+}
+
+type GetTokenRefreshLinkParams = {
+  onIdTokenExpired?: () => void,
+  onUpdateTokenFail?: ({ refreshToken: string, idToken: string }) => void,
+  onUpdateTokenSuccess?: (error?:Object) => void,
+}
+
+/**
+ * Interface of the createApolloLinks options
+ */
+type CreateApolloLinkParams = {
+  uri: string,
+  getAuthState?: () => AuthState,
```

Comment:
Why it's optional?

Reply:
Because this is needs only for the auth and refreshToken links